### PR TITLE
9.0 mig web widget one2many tags

### PR DESCRIPTION
**Pull request by susport**
Mig to 9.0 web_widget_one2many_tags

### CODE REVIEW

**Comment by jjscarafia** [comment]
I've try it locally but widget is displayed on the default way.
See image attached
seleccion_051

File: web_widget_one2many_tags/README.rst [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+Credits
+=======
+
+Contributors
+------------
+
+* Holger Brunn <hbrunn@therp.nl>
+* Iv??n Todorovich <ivan.todorovich@gmail.com>
+* Peter Hahn <peter.hahn@initos.com>
+*Adhoc SA <jjs@adhoc.com.ar>
```
Comment (hbrunn) [inline web_widget_one2many_tags/README.rst:50]: add a space

File: web_widget_one2many_tags/README.rst [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  @onchange handler.
+
+Credits
+=======
+
+Contributors
+------------
+
+* Holger Brunn <hbrunn@therp.nl>
+* Iv??n Todorovich <ivan.todorovich@gmail.com>
```
Comment (hbrunn) [inline web_widget_one2many_tags/README.rst:48]: @ivantodorovich's accented a didn't survive the replaying

File: web_widget_one2many_tags/static/src/js/web_widget_one2many_tags.js [hunk 3]
```diff
@@ -23,7 +24,7 @@ openerp.web_widget_one2many_tags = function(instance)
                     arrow: {
                         onArrowClick: function(e)
                         {
-                            var list_view = new instance.web.form.One2ManyListView(
+                            var list_view = new widget_one2many_tag.web.form.One2ManyListView(
```
Comment (hbrunn) [inline web_widget_one2many_tags/static/src/js/web_widget_one2many_tags.js:27]: no. all of them need to be required at the top of the file, then used in the code. Won't repeat this for every instance, but you basically need to change this in every changed line of code in this file

File: web_widget_one2many_tags/static/src/js/web_widget_one2many_tags.js [hunk 4]
```diff
@@ -209,4 +210,4 @@ openerp.web_widget_one2many_tags = function(instance)
             return this._super(record, column);
         },
     });
-}
+});
```
Comment (hbrunn) [inline web_widget_one2many_tags/static/src/js/web_widget_one2many_tags.js:213]: you need to return the classes this addons creates here in order to have then accessible for other addons

File: web_widget_one2many_tags/static/src/xml/web_widget_one2many_tags.xml [hunk 5]
```diff
@@ -1,5 +1,4 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <templates>
-    <t t-name="FieldOne2ManyTags" t-extend="FieldMany2ManyTags" />
-    <t t-name="FieldOne2ManyTag" t-extend="FieldMany2ManyTag" />
+     <t t-name="FieldOne2ManyTag" t-extend="FieldMany2ManyTag" />
```
Comment (hbrunn) [inline web_widget_one2many_tags/static/src/xml/web_widget_one2many_tags.xml:3]: the code above says we need both

File: web_widget_one2many_tags/views/templates.xml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+        <template id="assets_backend" name="web_widget_one2many_tags assets" inherit_id="web.assets_backend">
+            <xpath expr="." position="inside">
+                <script type="text/javascript" src="/web_widget_one2many_tags/static/src/js/web_widget_one2many_tags.js"></script>
+            </xpath>
+        </template>
```
Comment (hbrunn) [inline web_widget_one2many_tags/views/templates.xml:7]: watch your indentation

File: web_widget_one2many_tags/static/src/js/web_widget_one2many_tags.js [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+                                    widget_one2many_tag.web._t('New record');
+                            })
+                            .join(', ')
+                    );
+                });
+                column = _(column).extend({type: 'one2many_tags'});
+            }
+            return this._super(record, column);
+        },
+    });
```
Comment (hbrunn) [inline web_widget_one2many_tags/static/src/js/web_widget_one2many_tags.js:212]: you don't return anything, this means the classes defined here are not accessible to other addons

File: web_widget_one2many_tags/static/src/js/web_widget_one2many_tags.js [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+                ext: {
+                    itemManager: {
+                        itemToString: function(item) {
+                            return item.name;
+                        },
+                    },
+                    arrow: {
+                        onArrowClick: function(e)
+                        {
+                            var list_view = new widget_one2many_tag.web.form.One2ManyListView(
```
Comment (hbrunn) [inline web_widget_one2many_tags/static/src/js/web_widget_one2many_tags.js:27]: this a absolutely not the correct way to refer to objects that were accessed via `instance` in earlier versions. You need to `require` every object used above where you also require `Model`. And the canonical name then is the last dotted chunk.
You do this all over the place in this file, and need to change it everywhere

**Review by hbrunn — CHANGES_REQUESTED** [review]

**Comment by lasley** [comment]
Hi @susport - are you planning on continuing this PR?

**Comment by daramousk** [comment]
@hbrunn I need this ported to 10.0. Should I take over from here since there is no response?

**Comment by hbrunn** [comment]
in principle, yes, but check first if the user has submitted a CLA to the OCA. Without this, we can't merge the commits and then you'll be better off doing the migration yourself.

Closing this meanwhile for inactivity.

**Pull request closed without merging**